### PR TITLE
Showtimes Support

### PR DESCRIPTION
**Pull request by IanMitchell**
General style fixes included

### CODE REVIEW

**Comment by IanMitchell** [comment]
WIP

File: src/commands/showtimes.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const SHOWTIMES_URL = `${config.showtimes.server}/staff`;
+
+export class Showtimes extends Command {
+  message(from, to, text) {
+    const showtimesRegex = /^[.!](?:(?:st|(?:show(?:times)?))(?:\s))?(done|undone)(?:\s)(.+?)(?:\s!(\w+))?$/i;
+    const showtimes = text.match(showtimesRegex);
+
+    if (showtimes) {
+      return new Promise((resolve, reject) => {
+        const [, status, show, position] = showtimes;
```
Comment (IanMitchell) [inline src/commands/showtimes.js]: I dub this like "The Jukey Line"

File: src/arx7.js [hunk 2]
```diff
@@ -119,6 +121,26 @@ export class ARX7 {
           this.client.say(from, `I'm a bot! Contact [${admins}] for help`);
           resolve();
         }
+      }
+
+      const helpRegex = /^[.!](?:h(?:elp)?|man)(?:\s(.+))?$/i;
```
Comment (kunaldes) [inline src/arx7.js]: Hm... sometimes people join a channel and type !help or !rules to get help for the channel in general. Getting this message might not be very helpful.

File: src/arx7.js [hunk 3]
```diff
@@ -119,6 +121,26 @@ export class ARX7 {
...
+      const help = text.match(helpRegex);
+
+      if (help) {
+        if (help[1] === undefined) {
+          log(`Help request from ${from}`);
+          const str = `Syntax: .help [module]. Modules: ${this.moduleList()}.`;
+          this.client.say(to, str);
```
Comment (kunaldes) [inline src/arx7.js]: Is this the kind of thing we want to send to the channel or send as a notice?

File: src/arx7.js [hunk 4]
```diff
@@ -267,4 +289,14 @@ export class ARX7 {
...
   }
+
+  moduleList() {
+    let list = '';
+
+    this.commands.forEach(command => {
+      list += command.constructor.name.toLowerCase() + ', ';
+    });
+
+    return list.substring(0, list.length - 2);
```
Comment (kunaldes) [inline src/arx7.js]: This is kinda gross, and empty lists would make this weird.

maybe something like 

```javascript
let list = this.commands.map(command => { command.constructor.name.toLowerCase() });
return list.join(", ");
```

**Comment by kunaldes** [comment]
I don't really get promises lol.

But uh... mostly LGTM.

File: src/arx7.js (hunk 4 above)
Comment (IanMitchell) [inline src/arx7.js]: Thanks, was trying to remember what `join` was called the other night and was having trouble :+1:

File: src/arx7.js (hunk 3 above)
Comment (IanMitchell) [inline src/arx7.js]: I'm torn :wink: 

@BlueCrystal what do you think?

Comment (Fyurie) [inline src/arx7.js]: Usually messages meant for a single user that aren't useful to the wider public are just better sent as notices.

File: src/arx7.js (hunk 2 above)
Comment (IanMitchell) [inline src/arx7.js]: What would a better trigger be? I don't want to use .man because of Servrhe

File: src/arx7.js (hunk 3 above)
Comment (IanMitchell) [inline src/arx7.js]: Switched to notices

File: src/arx7.js (hunk 2 above)
Comment (Fyurie) [inline src/arx7.js]: might be slightly un-intuitive perhaps, but maybe .guide [command]?

**Pull request merged**